internal/gatewayapi/clustersettings.go

```diff
@@ -128,9 +129,19 @@ func buildClusterSettingsTimeout(policy egv1a1.ClusterSettings, traffic *ir.Traf
 			}
 		}
 
+		if pto.HTTP.RequestTimeout != nil {
+			d, err := time.ParseDuration(string(*pto.HTTP.RequestTimeout))
+			if err != nil {
+				errs = errors.Join(errs, fmt.Errorf("invalid RequestTimeout value %s", *pto.HTTP.RequestTimeout))
+			} else {
+				rt = ptr.To(metav1.Duration{Duration: d})
+			}
+		}
+
 		to.HTTP = &ir.HTTPTimeout{
 			ConnectionIdleTimeout: cit,
 			MaxConnectionDuration: mcd,
+			RequestTimeout:        rt,
```

Comment:
not seeing any changes in https://github.com/envoyproxy/gateway/blob/77c042cc146013bc731a27cbc4e7970610c31535/internal/gatewayapi/clustersettings.go#L149
can we ensure HTTPRoute Request Timeout takes precedence over BTP HTTP Request Timeout using a yaml test case?